tsl/src/continuous_aggs/invalidation.c

```diff
@@ -651,12 +793,34 @@ invalidation_process_hypertable_log(const ContinuousAgg *cagg,
 	invalidation_state_cleanup(&state);
 }
 
-void
+InvalidationStore *
 invalidation_process_cagg_log(const ContinuousAgg *cagg, const InternalTimeRange *refresh_window)
 {
 	CaggInvalidationState state;
+	InvalidationStore *store = NULL;
 
 	invalidation_state_init(&state, cagg, refresh_window);
+	state.invalidations = tuplestore_begin_heap(false, false, work_mem);
 	clear_cagg_invalidations_for_refresh(&state, refresh_window);
+
+	if (tuplestore_tuple_count(state.invalidations) == 0)
+		tuplestore_end(state.invalidations);
+	else
```

Comment:
Nit: usually good practice to have braces on both, or neither.
```suggestion
	if (tuplestore_tuple_count(state.invalidations) == 0)
	{
		tuplestore_end(state.invalidations);
	}
	else
```